tests/test_interpolation.py

```diff
@@ -709,3 +710,22 @@ def test_empty_stack() -> None:
     """
     with pytest.raises(GrammarParseError):
         grammar_parser.parse("ab}", lexer_mode="VALUE_MODE")
+
+
+@pytest.mark.parametrize("ref", ["missing", "invalid"])
+def test_invalid_intermediate_result_when_not_throwing(
+    ref: str, restore_resolvers: Any
+) -> None:
+    """
+    Check that the resolution of nested interpolations stops on missing / resolution failure.
+    """
+    OmegaConf.register_new_resolver("check_none", lambda x: x is None)
+    cfg = OmegaConf.create({"x": f"${{check_none:${{{ref}}}}}", "missing": "???"})
+    x_node = cfg._get_node("x")
+    assert isinstance(x_node, Node)
+    assert (
+        x_node._dereference_node(
+            throw_on_missing=False, throw_on_resolution_failure=False
+        )
+        is None
+    )
```

Comment:
I don't understand the purpose of the change this is testing.
Why do ywe want to ever get None from _dereference_node?

Reply:
I added a hopefully better explanation in dfea0a0
Before this PR, what would happen is that for instance `${invalid}` would be resolve to `None` (because `throw_on_resolution_failure` is False), and the resolver would be called with `None` as input (which in general could result in an arbitrary result)

Reply:
> Why do ywe want to ever get None from _dereference_node?

One example is `ValueNode._is_none()`